src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -1987,6 +1987,27 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
         //      actually fail. We need a way to gracefully fallback.
         _renderer->TriggerFontChange(newDpi, _desiredFont, _actualFont);
 
+        // If the actual font isn't what was requested...
+        if (_actualFont.GetFaceName() != _desiredFont.GetFaceName())
+        {
+            // Then warn the user that we picked something because we couldn't find their font.
+
+            // Format message with user's choice of font and the font that was chosen instead.
+            const winrt::hstring message{ fmt::format(std::wstring_view{ RS_(L"NoticeFontNotFound") }, _desiredFont.GetFaceName(), _actualFont.GetFaceName()) };
+
+            // Capture what we need to resume later.
+            [strongThis = get_strong(), message]() -> winrt::fire_and_forget {
+                // Pop the rest of this function to the tail of the UI thread
+                // Just in case someone was holding a lock when they called us and
+                // the handlers decide to do something that take another lock
+                // (like ShellExecute pumping our messaging thread...GH#7994)
+                co_await strongThis->Dispatcher();
+
+                auto noticeArgs = winrt::make_self<NoticeEventArgs>(NoticeLevel::Warning, message);

```

Comment:
Tiny nit:

You can use use `make` which returns the projected type directly. You can then std::move that into the constructor for your args type (and dispense with the `*`)

Doing it with the "self" variant introduces an intermediate `com_ptr` to the implementation type itself.